File: 栈和队列/检查替换后的词是否有效.java
package 栈和队列;

import java.util.ArrayDeque;
import java.util.Deque;

public class 检查替换后的词是否有效 {
    public boolean isValid(String s) {
        Deque<Character> stack = new ArrayDeque<>();

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == 'c') {
                if (stack.size() >= 2) {
                    char b = stack.pollLast();
                    if (b == 'b' && stack.peekLast() == 'a') stack.pollLast();
                    else return false;
                } else return false;
            } else {
                stack.add(c);
            }
        }
        return stack.isEmpty();
    }
}


class few{
    public static void main(String[] args){
        new 检查替换后的词是否有效().isValid("abcabcababcc");
    }
}

File: 栈和队列/检查替换后的词是否有效.c
//
// Created by 爽 on 2023/5/3.
//
#include  <stdbool.h>
#include <string.h>

bool isValid(char *s) {
    int n = strlen(s);
    int top = -1;
    char stack[n];//数组模拟栈，top模拟栈顶
    for (int i = 0; i < n; i++) {
        if (s[i] == 'c') {
            if (top >= 1 && stack[top] == 'b' && stack[top - 1] == 'a') {
                top -= 2;
            }else{
                return false;
            }
        } else {
            stack[++top] = s[i];
        }
    }
    return top == -1;
}